### DISCUSSION TITLE
Event propagation

### DISCUSSION THREAD

**Discussion by khinsen** [rabbibotton/clog#402] [Q&A] [answered]
Is there a way to manage event propagation in CLOG? Specifically, I would like to call `event.stopPropagation()` from a click event handler. But my handler doesn't have access to the JavaScript event object. Is there another way to do this?

**Comment by rabbibotton** [comment]
An extension just merged into clog from @mmontone and this from him
https://github.com/mmontone/clog-callbacks

Should do the trick.

**Comment by mmontone** [comment]
I'm using it to integrate a Javascript graph component: https://codeberg.org/mmontone/mold-desktop/src/branch/master/graph.lisp

![graph-callbacks](https://github.com/user-attachments/assets/f4304de8-2511-4459-8c18-27b87fd46770)

**Reply by aykaramba** [reply]
This is lovely. Thanks for sharing. I learned a lot from your code tests, I can't wait to dig into this. So much to learn!

**Comment by mmontone** [comment, marked as answer]
> Is there a way to manage event propagation in CLOG? Specifically, I would like to call `event.stopPropagation()` from a click event handler. But my handler doesn't have access to the JavaScript event object. Is there another way to do this?

Passing T in `:cancel-event` in CLOG event functions is not enough for that?

**Reply by khinsen** [reply]
Thanks, that does it!

But your callback extension looks very useful to me as well, for a different job. Thanks for that as well!
